### ISSUE TITLE
Consistent handling of optional columns

### ISSUE THREAD

**Issue by bifbof** [mie-lab/trackintel#491]
The Trackintel model has two types of columns that it accesses.
The required ones, without which you can't create a Trackintel object, and the optional ones mentioned in the documentation.
The handling of the optional columns is rather confusing
- some like `accuracy` are mentioned, but nowhere used with any function
- some like `trips` in tours are not mentioned at all, although they are returned every time by the `generate_tours` function
- no information is given about what type these optional types should be
- basically no tests are written for them, e.g. for io functions

We should treat them consistently and define: which ones are really used, their datatypes etc.
